calico-vpp-agent/watchers/peers_watcher.go

```diff
@@ -60,114 +51,56 @@ type ImpExpPol struct {
 	Exp *BGPPrefixesPolicyAndAssignment
 }
 
+// PeerWatcherHandler defines the interface for handling BGP peer updates
+type PeerWatcherHandler interface {
+	ProcessPeers(peers []calicov3.BGPPeer) error
+	OnSecretChanged(secretName string)
+}
+
 type PeerWatcher struct {
 	log      *logrus.Entry
 	clientv3 calicov3cli.Interface
 
-	// Subcomponent for accessing and watching secrets (that hold BGP passwords).
-	secretWatcher *secretWatcher
-
-	nodeStatesByName     map[string]common.LocalNodeSpec
-	peerWatcherEventChan chan any
-	BGPConf              *calicov3.BGPConfigurationSpec
 	watcher              watch.Interface
 	currentWatchRevision string
-}
 
-type bgpPeer struct {
-	AS            uint32
-	SweepFlag     bool
-	BGPPeerSpec   *calicov3.BGPPeerSpec
-	SecretChanged bool
+	handler       PeerWatcherHandler
+	secretWatcher *SecretWatcher
 }
 
-// selectsNode determines whether or not the selector mySelector
-// matches the labels on the given node.
-func selectsNode(mySelector string, n *common.LocalNodeSpec) (bool, error) {
-	// No node selector means that the selector matches the node.
-	if len(mySelector) == 0 {
-		return true, nil
-	}
-	// Check for valid selector syntax.
-	sel, err := selector.Parse(mySelector)
-	if err != nil {
-		return false, err
+func NewPeerWatcher(clientv3 calicov3cli.Interface, handler PeerWatcherHandler, secretWatcher *SecretWatcher, log *logrus.Entry) *PeerWatcher {
+	w := &PeerWatcher{
+		log:           log,
+		clientv3:      clientv3,
+		handler:       handler,
+		secretWatcher: secretWatcher,
```

Comment:
I think for simplicity, if we remove the need for handler's registration by introducing events
we could instantiate the `secretWatcher` directly here
i.e. 
```
secretWatcher: watchers.NewSecretWatcher(k8sclient, log.WithFields(logrus.Fields{"component": "secret-watcher"}))
```